run.ps1

```diff
@@ -0,0 +1,30 @@
+# set the base tools directory
+$toolsLocalPath = Join-Path $PSScriptRoot ".tools"
+$bootStrapperPath = Join-Path $toolsLocalPath "bootstrap.ps1"
+
+# if the boot-strapper script doesn't exist then download it
+if ((Test-Path $bootStrapperPath) -eq 0)
+{
+    if ((Test-Path $toolsLocalPath) -eq 0)
+    {
+        mkdir $toolsLocalPath | Out-Null
+    }
+
+    # download boot-strapper script
+    Invoke-WebRequest "https://raw.githubusercontent.com/dotnet/buildtools/master/bootstrap/bootstrap.ps1" -OutFile $bootStrapperPath
+}
+
+# now execute it
+Invoke-Expression "$bootStrapperPath (Get-Location) $toolsLocalPath" | Out-File (Join-Path (Get-Location) "bootstrap.log")
+if ($LastExitCode -ne 0)
```

Comment:
If I understand how lastexitcode works I believe it only has the exitcode of the last win32 program that was executed so it will not have the return code of the script you are calling directly. Although it is worth testing as I'm not 100% sure.